src/main/java/com/google/api/codegen/util/testing/TestValueGenerator.java

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util.testing;
+
+import com.google.api.codegen.util.Name;
+import com.google.api.tools.framework.model.TypeRef;
+
+import java.util.HashMap;
+
+/**
+ * A utility class used by the test generators which populates values for primitive fields.
+ *
+ * <p>The populated values will be unique (except for boolean values) and deterministic based on the
+ * populated fields and the sequence.
+ */
+public abstract class TestValueGenerator {
+  protected HashMap<Name, String> valueTable = new HashMap<>();
+
+  public String getAndStoreValue(TypeRef type, Name identifier) {
+    if (!valueTable.containsKey(identifier)) {
+      String value = generateValue(type, identifier);
+      while (type.getPrimitiveTypeName() != "bool" && valueTable.containsValue(value)) {
+        // If the value already exists regenerate using a deterministically different identifier.
+        identifier = identifier.join("1");
+        value = generateValue(type, identifier);
+      }
+      valueTable.put(identifier, generateValue(type, identifier));
+    }
+    return valueTable.get(identifier);
+  }
+
+  protected abstract String generateValue(TypeRef typeRef, Name identifier);
```

Comment:
When you are using inheritance, you are depending on polymorphic behavior. The polymorphic part is the generateValue call, and you provide 1 form so far for Java, JavaTestValueGenerator. Each new language will provide a new form called through the same method, making it polymorphic. 

Preferring composition over inheritance is not about flexibility in my mind, it is about https://en.wikipedia.org/wiki/Loose_coupling . When you use inheritance, the derived class knows the details of the class it inherits from. When you use composition, the class implementing the interface has no knowledge at all of the class calling that interface. That interface also becomes available to other classes independent of the original code calling that interface. 

This loose coupling makes the system easier to change and evolve over time, because you can mix and match the independent components. As the wikipedia article on composition over inheritance states, "This approach more easily accommodates future requirements changes that would otherwise require a complete restructuring of business-domain classes in the inheritance model." If you use too much inheritance, you don't get pieces, you get a bunch of things tied together. This makes the system more complex to understand, even if it technically has fewer classes. Number of classes is not the end-all metric for determining how complex a system is.